### ISSUE TITLE
API calls are all apparently broken in 2.14.0 (renderer format not valid)

### ISSUE THREAD

**Issue by propertunist** [piwik/piwik#8333]
as can be seen in this thread: http://forum.piwik.org/read.php?2,128007
the standard method of making API calls is presently broken.

**Comment by mattab** [comment]
How do you reproduce the issue? the web API request are working on demo eg. http://demo.piwik.org/index.php?module=API&method=Actions.getPageUrls&format=json&idSite=7&period=day&date=today

**Comment by propertunist** [comment]
i only have one API call that i am making outside of piwik (i.e. from my own app's code). i used the format of API request that is suggested in the piwik documentation for PHP. in this example i am passing in:

```
$request = new Request ('
module=API
&method=UserCountry.getLocationFromIP
&ip=' . $current_IP . '
&format=json
&token_auth=' . $piwik_secret
);
```

**Comment by mattab** [comment]
I think this duplicates https://github.com/piwik/piwik/issues/8311 - please reopen if it's a different issue

**Comment by braekling** [comment]
Hi @mattab - I applied the patch, but now I run into this:

```
Fatal error: Uncaught exception 'Piwik\Container\ContainerDoesNotExistException' with message 'The root container has not been created yet.' in /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php:40 Stack trace: #0 /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php(80): Piwik\Container\StaticContainer::getContainer() #1 /var/www/vhosts/.../httpdocs/stats/core/FrontController.php(206): Piwik\Container\StaticContainer::get('path.tmp') #2 /var/www/vhosts/.../httpdocs/wp-content/plugins/wp-piwik/classes/WP_Piwik/Request/Php.php(31): Piwik\FrontController->init() #3 /var/www/vhosts/.../httpdocs/wp-content/plugins/wp-piwik/classes/WP_Piwik/Request/Php.php(14): WP_Piwik\Request\Php->call('method=VisitsSu...', 'http://www.brae...', 'module=API&form...') #4 /var/www/vhosts/.../httpdocs/wp-content/plugins/wp-piwik/classes/WP_Piwik/Request.php(63): WP_Piwik\Request\Php->request('method=VisitsSu...') #5 /var/www/vhosts/.../httpdocs/wp- in /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php on line 40

Fatal error: Uncaught exception 'Piwik\Container\ContainerDoesNotExistException' with message 'The root container has not been created yet.' in /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php:40 Stack trace: #0 /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php(80): Piwik\Container\StaticContainer::getContainer() #1 /var/www/vhosts/.../httpdocs/stats/core/Config.php(62): Piwik\Container\StaticContainer::get('Piwik\Config') #2 /var/www/vhosts/.../httpdocs/stats/core/Url.php(347): Piwik\Config::getInstance() #3 /var/www/vhosts/.../httpdocs/stats/core/Url.php(67): Piwik\Url::getCurrentHost() #4 /var/www/vhosts/.../httpdocs/stats/core/FrontController.php(87): Piwik\Url::getCurrentUrl() #5 /var/www/vhosts/.../httpdocs/stats/core/FrontController.php(181): Piwik\FrontController->dispatch('CorePluginsAdmi...', 'safemode', Array) #6 [internal function]: Piwik\FrontController::triggerSafeModeWhenError() #7 {main} thrown in /var/www/vhosts/.../httpdocs/stats/core/Container/StaticContainer.php on line 40
```

Maybe I miss something? This is how I init the FrontController:

```
if (file_exists(PIWIK_INCLUDE_PATH . "/index.php"))
   require_once PIWIK_INCLUDE_PATH . "/index.php";
if (file_exists(PIWIK_INCLUDE_PATH . "/core/API/Request.php"))
   require_once PIWIK_INCLUDE_PATH . "/core/API/Request.php";
if (class_exists('Piwik\FrontController'))
   \Piwik\FrontController::getInstance()->init();
```

I'm not able to get WP-Piwik working again since Piwik 2.14.0.

**Comment by propertunist** [comment]
this issue has not been fixed for me in 2.14.1.
i have run the upgrade and i still see exactly the same fatal errors when running an API call. the 2.14 workaround still changes the fatal error from the 'root container has not been created yet' message to the 'renderer format not valid: 'json'' message. but ultimately the outcome remains - fatal error.
